tests/unit/test_run.py

```diff
@@ -1393,6 +1393,67 @@ def test_process_over_without_na_override(scm_run):
     assert len(res) == 0  # This result is incorrect due to the disabling na_override
 
 
+def test_process_over_as_run(scm_run):
```

Comment:
I'd suggest adding a test of the error if you `process_over` returns a series but `as_run` is True, it's not clear to me what would happen there (nor if it would make any sense to the user).

Reply:
It correctly was cast to an ScmRun, but with a garbage time value. New implementation raises a ValueError in that case

Reply:
beauty